### PR TITLE
Fix TE EP capacity sizing for gradient accumulation

### PR DESCRIPTION
**Pull request by abhinavgoel95**
# Description

Fix TransformerEngine EP buffer capacity sizing when gradient accumulation is enabled.

Previously, EP bootstrap used the entire accumulated batch to calculate `max_tokens_per_rank`. Each MoE invocation processes only one microbatch, so this unnecessarily increased receive-buffer and staging-buffer capacity with GA, potentially causing OOMs.

This change:
- Divides the loaded batch size by `gradient_accumulation_steps`.
- Preserves sufficient capacity for model initialization and larger evaluation batches.
- Validates GA and FSDP/EP batch divisibility.
- Leaves TE’s receive-capacity calculation and routing headroom unchanged.

For a workload with 6 sequences per GPU and sequence length 4096, the token bound now remains 24,576 tokens/rank regardless of GA.

This addresses EP overprovisioning, not separate XLA buffer-lifetime memory growth.

# Tests

- Added and passed 10 CPU unit tests with mocked TE bootstrap, covering:
  - GA=1, 2, 3, 16, and 32.
  - Larger, smaller, and disabled evaluation batches.
  - Expanded loaded batches and initialization sizing.
  - Invalid batch sizes and GA values.
  - Bootstrap caching and non-TE configurations.
- Passed Pyink formatting and `git diff --check`.
- Tested the same sizing change on a 16-GPU small DeepSeek workload with 5 layers, 128 experts, EP=4, and FSDP=4. GA=2, GA=3, and GA=16 all completed 15 training steps. EP receive capacity remained 393,216 slots/rank.
- Workload logs are internal and are not linked publicly.

Run the unit tests from the repository root in a MaxText development environment:

    DECOUPLE_GCLOUD=TRUE PYTHONPATH=src \
      python -m unittest discover \
      -s tests/unit -p te_moe_bootstrap_test.py -v
      
# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [ x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ x] I have run end-to-end tests tests and provided workload links above if applicable.
- [ x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/utils/max_utils.py [hunk 1]
```diff
@@ -1306,7 +1306,17 @@ def maybe_bootstrap_te_moe(config, mesh, shaped_batch):
   ep_size = mesh.shape.get(ep_axis, 1)
   fsdp_size = mesh.shape.get(fsdp_axis, 1)
 
-  batch_size, sequence_length = shaped_batch["inputs"].shape[:2]
+  loaded_batch_size, sequence_length = shaped_batch["inputs"].shape[:2]
+  ga_steps = config.gradient_accumulation_steps
+  if ga_steps < 1 or loaded_batch_size % ga_steps != 0:
```
Comment (gobbleturk) [inline src/maxtext/utils/max_utils.py]: when would ga_steps be < 1?

**Review by gobbleturk — APPROVED** [review]
